diff --git a/node_modules/@tldraw/tlschema/dist-esm/index.d.mts b/node_modules/@tldraw/tlschema/dist-esm/index.d.mts
index a57433b..3cf80a1 100644
--- a/node_modules/@tldraw/tlschema/dist-esm/index.d.mts
+++ b/node_modules/@tldraw/tlschema/dist-esm/index.d.mts
@@ -50,6 +50,7 @@ export declare const arrowShapeVersions: {
     readonly AddIsPrecise: "com.tldraw.shape.arrow/2";
     readonly AddLabelColor: "com.tldraw.shape.arrow/1";
     readonly AddLabelPosition: "com.tldraw.shape.arrow/3";
+    readonly AddRichText: "com.tldraw.shape.arrow/7";
     readonly AddScale: "com.tldraw.shape.arrow/5";
     readonly ExtractBindings: "com.tldraw.shape.arrow/4";
 };
@@ -807,7 +808,7 @@ export declare interface TLArrowShapeProps {
     start: VecModel;
     end: VecModel;
     bend: number;
-    text: string;
+    richText: TLRichText;
     labelPosition: number;
     scale: number;
     elbowMidPoint: number;
diff --git a/node_modules/@tldraw/tlschema/dist-esm/shapes/TLArrowShape.mjs b/node_modules/@tldraw/tlschema/dist-esm/shapes/TLArrowShape.mjs
index 8ca9bfa..790f6c7 100644
--- a/node_modules/@tldraw/tlschema/dist-esm/shapes/TLArrowShape.mjs
+++ b/node_modules/@tldraw/tlschema/dist-esm/shapes/TLArrowShape.mjs
@@ -1,5 +1,6 @@
 import { createMigrationSequence } from "@tldraw/store";
 import { T } from "@tldraw/validate";
+import { richTextValidator, toRichText } from "../misc/TLRichText.mjs";
 import { vecModelValidator } from "../misc/geometry-types.mjs";
 import { createBindingId } from "../records/TLBinding.mjs";
 import { createShapePropsMigrationIds } from "../records/TLShape.mjs";
@@ -50,7 +51,7 @@ const arrowShapeProps = {
   start: vecModelValidator,
   end: vecModelValidator,
   bend: T.number,
-  text: T.string,
+  richText: richTextValidator,
   labelPosition: T.number,
   scale: T.nonZeroNumber,
   elbowMidPoint: T.number
@@ -61,7 +62,8 @@ const arrowShapeVersions = createShapePropsMigrationIds("arrow", {
   AddLabelPosition: 3,
   ExtractBindings: 4,
   AddScale: 5,
-  AddElbow: 6
+  AddElbow: 6,
+  AddRichText: 7
 });
 function propsMigration(migration) {
   return createPropsMigration("shape", "arrow", migration);
@@ -184,6 +186,17 @@ const arrowShapeMigrations = createMigrationSequence({
         delete props.kind;
         delete props.elbowMidPoint;
       }
+    }),
+    propsMigration({
+      id: arrowShapeVersions.AddRichText,
+      up: (props) => {
+        props.richText = toRichText(props.text);
+        delete props.text;
+      }
+      // N.B. Explicitly no down state so that we force clients to update.
+      // down: (props) => {
+      // 	delete props.richText
+      // },
     })
   ]
 });
diff --git a/node_modules/@tldraw/tlschema/src/migrations.test.ts b/node_modules/@tldraw/tlschema/src/migrations.test.ts
index 18f0c5f..74b3f69 100644
--- a/node_modules/@tldraw/tlschema/src/migrations.test.ts
+++ b/node_modules/@tldraw/tlschema/src/migrations.test.ts
@@ -1385,6 +1385,7 @@ describe('Add rich text', () => {
 		['text shape', getTestMigration(textShapeVersions.AddRichText)],
 		['geo shape', getTestMigration(geoShapeVersions.AddRichText)],
 		['note shape', getTestMigration(noteShapeVersions.AddRichText)],
+		['arrow shape', getTestMigration(arrowShapeVersions.AddRichText)],
 	] as const
 
 	for (const [shapeName, { up }] of migrations) {
diff --git a/node_modules/@tldraw/tlschema/src/shapes/TLArrowShape.ts b/node_modules/@tldraw/tlschema/src/shapes/TLArrowShape.ts
index 02c0a77..b2f087f 100644
--- a/node_modules/@tldraw/tlschema/src/shapes/TLArrowShape.ts
+++ b/node_modules/@tldraw/tlschema/src/shapes/TLArrowShape.ts
@@ -1,5 +1,6 @@
 import { createMigrationSequence } from '@tldraw/store'
 import { T } from '@tldraw/validate'
+import { TLRichText, richTextValidator, toRichText } from '../misc/TLRichText'
 import { VecModel, vecModelValidator } from '../misc/geometry-types'
 import { createBindingId } from '../records/TLBinding'
 import { TLShapeId, createShapePropsMigrationIds } from '../records/TLShape'
@@ -67,7 +68,7 @@ export interface TLArrowShapeProps {
 	start: VecModel
 	end: VecModel
 	bend: number
-	text: string
+	richText: TLRichText
 	labelPosition: number
 	scale: number
 	elbowMidPoint: number
@@ -90,7 +91,7 @@ export const arrowShapeProps: RecordProps<TLArrowShape> = {
 	start: vecModelValidator,
 	end: vecModelValidator,
 	bend: T.number,
-	text: T.string,
+	richText: richTextValidator,
 	labelPosition: T.number,
 	scale: T.nonZeroNumber,
 	elbowMidPoint: T.number,
@@ -104,6 +105,7 @@ export const arrowShapeVersions = createShapePropsMigrationIds('arrow', {
 	ExtractBindings: 4,
 	AddScale: 5,
 	AddElbow: 6,
+	AddRichText: 7,
 })
 
 function propsMigration(migration: TLPropsMigration) {
@@ -254,5 +256,16 @@ export const arrowShapeMigrations = createMigrationSequence({
 				delete props.elbowMidPoint
 			},
 		}),
+		propsMigration({
+			id: arrowShapeVersions.AddRichText,
+			up: (props) => {
+				props.richText = toRichText(props.text)
+				delete props.text
+			},
+			// N.B. Explicitly no down state so that we force clients to update.
+			// down: (props) => {
+			// 	delete props.richText
+			// },
+		}),
 	],
 })
