diff --git a/node_modules/tldraw/dist-cjs/lib/TldrawImage.js b/node_modules/tldraw/dist-cjs/lib/TldrawImage.js
index 2b7d84d..24c4fc7 100644
--- a/node_modules/tldraw/dist-cjs/lib/TldrawImage.js
+++ b/node_modules/tldraw/dist-cjs/lib/TldrawImage.js
@@ -49,7 +49,7 @@ const TldrawImage = (0, import_react.memo)(function TldrawImage2(props) {
     () => (0, import_editor.mergeArraysAndReplaceDefaults)("type", _bindingUtils, import_defaultBindingUtils.defaultBindingUtils),
     [_bindingUtils]
   );
-  const store = (0, import_editor.useTLStore)({ snapshot: props.snapshot, shapeUtils: shapeUtilsWithDefaults });
+  const store = (0, import_editor.useTLStore)({ assets: props.assets, snapshot: props.snapshot, shapeUtils: shapeUtilsWithDefaults });
   const {
     pageId,
     bounds,
diff --git a/node_modules/tldraw/dist-esm/lib/TldrawImage.mjs b/node_modules/tldraw/dist-esm/lib/TldrawImage.mjs
index 781fc92..bddce7d 100644
--- a/node_modules/tldraw/dist-esm/lib/TldrawImage.mjs
+++ b/node_modules/tldraw/dist-esm/lib/TldrawImage.mjs
@@ -31,7 +31,7 @@ const TldrawImage = memo(function TldrawImage2(props) {
     () => mergeArraysAndReplaceDefaults("type", _bindingUtils, defaultBindingUtils),
     [_bindingUtils]
   );
-  const store = useTLStore({ snapshot: props.snapshot, shapeUtils: shapeUtilsWithDefaults });
+  const store = useTLStore({ assets: props.assets, snapshot: props.snapshot, shapeUtils: shapeUtilsWithDefaults });
   const {
     pageId,
     bounds,
diff --git a/node_modules/tldraw/src/lib/TldrawImage.tsx b/node_modules/tldraw/src/lib/TldrawImage.tsx
index c082b1f..71e3915 100644
--- a/node_modules/tldraw/src/lib/TldrawImage.tsx
+++ b/node_modules/tldraw/src/lib/TldrawImage.tsx
@@ -2,6 +2,7 @@ import {
 	Editor,
 	TLAnyBindingUtilConstructor,
 	TLAnyShapeUtilConstructor,
+	TLAssetStore,
 	TLEditorSnapshot,
 	TLImageExportOptions,
 	TLPageId,
@@ -48,6 +49,10 @@ export interface TldrawImageProps extends TLImageExportOptions {
 	 * The license key.
 	 */
 	licenseKey?: string
+	/**
+	 * The asset store to use for resolving assets.
+	 */
+	assets?: TLAssetStore
 	/**
 	 * Asset URL overrides.
 	 */
@@ -105,7 +110,7 @@ export const TldrawImage = memo(function TldrawImage(props: TldrawImageProps) {
 		() => mergeArraysAndReplaceDefaults('type', _bindingUtils, defaultBindingUtils),
 		[_bindingUtils]
 	)
-	const store = useTLStore({ snapshot: props.snapshot, shapeUtils: shapeUtilsWithDefaults })
+	const store = useTLStore({ assets: props.assets, snapshot: props.snapshot, shapeUtils: shapeUtilsWithDefaults })
 
 	const {
 		pageId,
